### PR TITLE
Add Kotlin listings for OkHttp recipes doc

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ markdown_extensions:
   - pymdownx.magiclink
   - pymdownx.smartsymbols
   - pymdownx.superfences
+  - pymdownx.tilde
```
Comment (swankjesse) [inline mkdocs.yml]: necessary for `~~foo~~` to render foo in strikeout, which we do on our TLS configuration history page

File: samples/guide/src/main/java/okhttp3/recipes/kt/Authenticate.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          val credential = Credentials.basic("jesse", "password1")
+          return response.request.newBuilder()
+              .header("Authorization", credential)
+              .build()
+        }
+      })
+      .build()
+
+  fun run() {
+    val request = Request.Builder()
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/kt/Authenticate.kt]: Any idea what breaks github highlighting after this point?

image

File: samples/guide/src/main/java/okhttp3/recipes/kt/ParseResponseWithMoshi.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      val gist = gistJsonAdapter.fromJson(response.body!!.source())
+
+      for ((key, value) in gist!!.files!!) {
+        println(key)
+        println(value.content)
+      }
+    }
+  }
+
+  @JsonClass(generateAdapter = true)
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/kt/ParseResponseWithMoshi.kt]: nice example

File: samples/guide/src/main/java/okhttp3/recipes/kt/ParseResponseWithMoshi.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ */
+package okhttp3.recipes.kt
+
+import com.squareup.moshi.JsonClass
+import com.squareup.moshi.Moshi
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.IOException
+
+class ParseResponseWithMoshi {
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/kt/ParseResponseWithMoshi.kt]: You wanted these examples completely self contained, so maybe add the kapt line from the gradle build file as a class comment?

Comment (swankjesse) [reply samples/guide/src/main/java/okhttp3/recipes/kt/ParseResponseWithMoshi.kt]: I do, but I also don’t want to go deep into explaining Moshi stuff. I think it’ll become clear if they try and Moshi tells you you forgot to run the generator.

File: samples/guide/src/main/java/okhttp3/recipes/kt/Authenticate.kt (hunk 2 above)
Comment (swankjesse) [reply samples/guide/src/main/java/okhttp3/recipes/kt/Authenticate.kt]: It usually renders comments in grey, so it’s got to be confused when it gets there.

**Pull request merged**